eng/common/TestResources/New-TestResources.ps1

```diff
@@ -11,14 +11,14 @@
 [CmdletBinding(DefaultParameterSetName = 'Default', SupportsShouldProcess = $true, ConfirmImpact = 'Medium')]
 param (
     # Limit $BaseName to enough characters to be under limit plus prefixes, and https://docs.microsoft.com/azure/architecture/best-practices/resource-naming.
-    [Parameter(Position = 0)]
+    [Parameter()]
     [ValidatePattern('^[-a-zA-Z0-9\.\(\)_]{0,80}(?<=[a-zA-Z0-9\(\)])$')]
     [string] $BaseName,
 
     [ValidatePattern('^[-\w\._\(\)]+$')]
     [string] $ResourceGroupName,
 
-    [Parameter(Mandatory = $true)]
+    [Parameter(Mandatory = $true, Position = 0)]
```

Comment:
This is a breaking change. You moved the first unnamed parameter position. Leave them both named.

Reply:
Do you think it would actually break anyone?

Reply:
I think simplifying the script is worth it.

Reply:
Can't say. Do you think specifying `-ServiceDirectory` is too much to type especially compared to everything they had to type before, whether using the script or az CLI directly?

I suppose, at the very least, both were mandatory before so if they did expect this to be `$BaseName`, they'd likely get an error that "sdk/Whatever" doesn't exist.

Reply:
>Can't say. Do you think specifying -ServiceDirectory is too much to type especially compared to everything they had to type before, whether using the script or az CLI directly?

Not too much comparatively, but if we can do even better and it's pretty safe I would like to try.

Reply:
If you know you are introducing a breaking change, make sure you go to the sync PRs that get created on changes in this directory and trigger some of the live test pipelines against them just to make sure that they continue to function.

Reply:
Great point @mitchdenny, I'll do that.

Reply:
Where are you defaulting the ServiceDirectory? Isn't it needed in order to know which arm template to deploy?

Reply:
I don't default the ServiceDirectory only the base name. ServiceDirectory is not the only required parameter and I made it positional as well.